src/itest/kotlin/nl/info/zac/itest/ZaakKoppelenRestServiceTest.kt

```diff
@@ -8,36 +8,214 @@ import io.github.oshai.kotlinlogging.KotlinLogging
 import io.kotest.core.spec.Order
 import io.kotest.core.spec.style.BehaviorSpec
 import io.kotest.matchers.shouldBe
-import nl.info.zac.itest.client.ZacClient
+import nl.info.zac.itest.client.ItestHttpClient
 import nl.info.zac.itest.config.ItestConfiguration.HTTP_STATUS_OK
-import nl.info.zac.itest.config.ItestConfiguration.TEST_SPEC_ORDER_AFTER_ZAAK_CREATED
+import nl.info.zac.itest.config.ItestConfiguration.TEST_SPEC_ORDER_AFTER_SEARCH
+import nl.info.zac.itest.config.ItestConfiguration.ZAAK_MANUAL_2020_01_IDENTIFICATION
+import nl.info.zac.itest.config.ItestConfiguration.ZAAK_MANUAL_2024_01_IDENTIFICATION
+import nl.info.zac.itest.config.ItestConfiguration.ZAAK_PRODUCTAANVRAAG_1_IDENTIFICATION
+import nl.info.zac.itest.config.ItestConfiguration.ZAC_API_URI
 import nl.info.zac.itest.config.ItestConfiguration.zaakProductaanvraag1Uuid
+import nl.info.zac.itest.util.shouldEqualJsonIgnoringOrderAndExtraneousFields
+import org.json.JSONObject
+import java.util.UUID
 
 private const val ROWS_DEFAULT = 10
 private const val PAGE_DEFAULT = 0
+private const val ZOEK_ZAAK_IDENTIFIER = "ZAAK-"
 
 /**
  * This test assumes a zaak has been created in a previously run test.
  */
-@Order(TEST_SPEC_ORDER_AFTER_ZAAK_CREATED)
+@Order(TEST_SPEC_ORDER_AFTER_SEARCH)
 @Suppress("LargeClass")
 class ZaakKoppelenRestServiceTest : BehaviorSpec({
-    val zacClient = ZacClient()
+    val itestHttpClient = ItestHttpClient()
     val logger = KotlinLogging.logger {}
+    lateinit var zaakUUID: UUID
 
-    Given("ZAC Docker container is running and zaakafhandelparameters have been created") {
-        When("the searching for linkable cases happy path") {
-            val response = zacClient.findLinkableCases(
-                zaakProductaanvraag1Uuid,
-                "ZAAC_2000345",
-                "HOOFDZAAK",
-                PAGE_DEFAULT,
-                ROWS_DEFAULT
+    Given("ZAC Docker container is running and the zaakafhandelparameters have been created") {
+        itestHttpClient.performGetRequest(
+            "$ZAC_API_URI/zaken/zaak/id/$ZAAK_MANUAL_2024_01_IDENTIFICATION"
+        ).use { getZaakResponse ->
+            val responseBody = getZaakResponse.body!!.string()
+            logger.info { "Response: $responseBody" }
+            with(JSONObject(responseBody)) {
+                zaakUUID = getString("uuid").let(UUID::fromString)
+            }
+        }
+
+        When("searching for a HOOFDZAAK linkable zaken for indienen-aansprakelijkstelling-behandelen zaak") {
```

Comment:
`indienen-aansprakelijkstelling-behandelen zaak` is incorrect it seems to me? I see that you are searching on the term `ZAAK-`. Nothing to do with the indienen-aansprakelijkstelling-behandelen zaaktype?

Reply:
Why actually do you search on all zaken here (using `ZAAK-`)? Is it not easier and clearer to make the search more refined. E.g. search on `ZAAK-2000` or so?

Reply:
Maybe better even would be if you create two zaken here yourself, and then only search on those zaken so that this test becomes a little bit more independent from other tests. But that's too much work for now so forget about that. Still good to keep in mind that these tests need to be maintained and the more dependent they are on other tests, the more difficult this becomes.